packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -977,8 +977,8 @@ export class IndexedDbPersistence implements Persistence {
         // to make sure it gets a chance to run.
         this.markClientZombied();
 
-        if (isSafari() && navigator.appVersion.match(/Version\/1[45]/)) {
-          // On Safari 14 and 15, we do not run any cleanup actions as it might
+        if (isSafari() && (navigator.appVersion.match(/(?:Version|Mobile)\/1[456]/) || navigator.userAgent.match(/(?:Version|Mobile)\/1[456]/))) {
```

Comment:
Please factor out the regular expression into a local variable, to avoid duplicating it.

e.g. `const versionRegex = /(?:Version|Mobile)\/1[456]/;`